numba/tests/test_tuples.py

```diff
@@ -609,6 +609,178 @@ def test_index(self):
+        self.assertEqual(d1, expected1)
+        self.assertEqual(d2, tuple(range(3, 6)))
+        self.assertEqual(e1, expected1)
+        self.assertEqual(e2, expected1)
+        self.assertEqual(f1, tuple([(True, True), (True, False)]))
+        self.assertEqual(f2, tuple([(True, True), (True, False)]))
+
+    def test_make_tuple_excess_data(self):
+        from numba.targets.tupleobj import make_tuple
+    
+        @njit
+        def foo():
+            a = make_tuple(3, list(range(6)))
+            b = make_tuple(3, range(6))
+            c = make_tuple(3, zip(range(0, 6), range(6, 12)))
+            it = iter(range(6))
+            d = make_tuple(3, it)
+            arr = np.array(list(range(6)))
+            e = make_tuple(3, arr)
+            f = make_tuple(2, [(True, True), (True, False), (False, False)])
+            return a, b, c, d, e, f
+
+        a, b, c, d, e, f = foo()
+        expected1 = tuple(range(3))
+        self.assertEqual(a, expected1)
+        self.assertEqual(b, expected1)
+        self.assertEqual(c, tuple(zip(range(0, 3), range(6, 9))))
+        self.assertEqual(d, expected1)
+        self.assertEqual(e, expected1)
+        self.assertEqual(f, tuple([(True, True), (True, False)]))
+
+    def test_make_tuple_insufficient_data(self):
+        from numba.targets.tupleobj import make_tuple
+    
+        @njit
+        def foo():
+            a = make_tuple(3, [1])
+            return a
+
+        with self.assertRaises(ValueError) as raises:
+            a = foo()
+        expected = "shorter than specified"
+        self.assertIn(expected, str(raises.exception))
+
+        @njit
+        def foo():
+            it = iter([1])
+            a = make_tuple(3, it)
+            return a
+
+        with self.assertRaises(ValueError) as raises:
+            a = foo()
+        expected = "shorter than specified"
+        self.assertIn(expected, str(raises.exception))
+
+    def test_make_tuple_invalid_arguments(self):
+        from numba.targets.tupleobj import make_tuple
+    
+        @njit
+        def foo(x):
+            a = make_tuple(x, [1])
+            return a
+
+        with self.assertRaises(errors.TypingError) as raises:
+            a = foo(3)
+        expected = "must be a constant integer"
+        self.assertIn(expected, str(raises.exception))
+
+        @njit
+        def foo():
+            a = make_tuple(3, 1)
+            return a
+
+        with self.assertRaises(errors.TypingError) as raises:
+            a = foo()
+        expected = "must be an iterable"
+        self.assertIn(expected, str(raises.exception))
+
+        @njit
+        def foo():
+            arr = np.array([[1], [2]])
+            a = make_tuple(3, arr)
+            return a
+
+        with self.assertRaises(errors.TypingError) as raises:
+            a = foo()
+        expected = "must be 1d"
+        self.assertIn(expected, str(raises.exception))
+
+    def test_to_fixed_tuple(self):
```

Comment:
Not sure about this test name, was it perhaps inadvertently copied from the now removed tests?

Reply:
@esc It was indeed copy-pasted. I originally kept it because it contains some test-cases that I did not test explicitly elsewhere. I will copy its contents to a more appropriate test case.